sphinx_immaterial/local_mathjax.py

```diff
@@ -0,0 +1,64 @@
+from pathlib import Path
+import shutil
+from typing import Dict, Any, cast
+from sphinx.application import Sphinx
+from sphinx.config import Config
+from sphinx.errors import ExtensionError
+from sphinx.ext.mathjax import MATHJAX_URL, MathDomain
+
+
+def copy_mathjax_dist(
+    app: Sphinx,
+    pagename: str,
+    templatename: str,
+    context: Dict[str, Any],
+    event_arg: Any,
+) -> None:
+    if (
+        app.builder.format != "html"
+        or app.builder.math_renderer_name != "mathjax"  # type: ignore[attr-defined]
+    ):
+        return
+    if not app.config.mathjax_path:
+        raise ExtensionError(
+            "mathjax_path config value must be set for the mathjax extension to work"
+        )
+    if (
+        "mathjax_path" in app.config
+        and app.config["mathjax_path"] != "mathjax/tex-mml-chtml.js"
+    ):
+        return  # not using theme's cache of mathjax dist
+
+    domain = cast(MathDomain, app.env.get_domain("math"))
+    if app.registry.html_assets_policy == "always" or domain.has_equations(pagename):
+        # copy mathjax fonts only if equations exists (and not already done)
+        dst = Path(app.outdir, "_static", "mathjax")
+        if Path(app.outdir, "_static", dst).exists():
```

Comment:
I think rather than checking if the path exists in the output dir, it would be better to just always copy the files if mathjax is being used, but not more than once per build.

We could accomplish that by adding a hook to the `env-updated` event.

Reply:
I only used the `html-page-context` event because `domain.has_equations(pagename)` seemed vital. Most of this function's beginnings were straight copied from the `mathjax.install_mathjax()` function. Though I think the `env-updated` event can be used similarly, I just have to figure out how it could fit with the `domain.has_equations()` call.

Reply:
Oh looks like the `docname` param to `domain.has_equations` is optional... commiting shortly.

Reply:
done